MyFirstSlideshow/app/src/main/res/layout/activity_main.xml

```diff
@@ -4,11 +4,87 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     tools:context="com.charles.myfirstslideshow.MainActivity">
+    <LinearLayout
```

Comment:
activity_main.xml:line7 - It is better to use constaint layout.rather than linear where we have to add several nested layout which is creating a complex herachy and taking much rendering time

Reply:
activity_main.xml:line7- specify the orientation of LinearLayout, you can use constraint layout also